server/metadata/namespace.go

```diff
@@ -0,0 +1,38 @@
+package metadata
+
+import (
+	"context"
+	"fmt"
+
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/transaction"
+)
+
+// NamespaceExtractor - extract the namespace from context
+type NamespaceExtractor interface {
+	Extract(ctx context.Context) (Namespace, error)
+}
+
+type NamespaceCtxKey struct {
+}
+
+type AccessTokenNamespaceExtractor struct {
+	TenantManager      *TenantManager
+	Transactionmanager *transaction.Manager
+}
+
+func (tokenNamespaceExtractor *AccessTokenNamespaceExtractor) Extract(ctx context.Context) (Namespace, error) {
```

Comment:
Once we move the class outside of this package then `Extract` should only be responsible for extracting the namespace name. In sessions.go, you'll then use this name to ask the tenant if it exists(if we need to do strict checking) and returns an error if it doesn't exist.

Reply:
there are admin apis, which will be at cluster level and namespace will be nil for those, do you recommend setting default namespace for those in context from middleware. This way we never expect nil namespace in context in sessions.go?

Reply:
That makes sense, we can set the default from outside. Then we can do strict check in the sessions.go

Reply:
👍